### ISSUE TITLE
Missing information in tutorial supporting broad range of models

### ISSUE THREAD

**Issue by kirk86** [huggingface/transformers#30054]
### System Info

transformers version: 4.39.2
python 3.12
platform linux

### Who can help?

@stevhliu @MKhalusova 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The [example here](https://github.com/huggingface/transformers/tree/main/examples/pytorch/summarization) states that among the supported architectures for summmarization are `Bart` and `Pegasus`.

A couple of clarifying questions.

1. ~~Does the data need to the have the same format as for `T5`, i.e. using format such as `prefix + actual text data`?~~ It seems the answer is no. At least that's what I get from the examples.
2. ~~The models such as `Bart` and `Pegasus` do not implement `ignore_index` in cross entropy to ignore padding `-100` that is being added automatically by the `DataCollatorForSeq2Seq` in contrary to the `T5` model which does.~~ Seems like when calling cross entropy loss from pytorch comes with default value of `ignore_index=-100` already predefined. I would prefer a more explicit behaviour to make sure the user understands what's going on in the code instead of second guessing default values which might change in the future

Are we sure that if we follow the example linked above and make a simple MWE setting using `ForSeq2Seq` trainer, args, models, collator, that we won't get any errors? I mean the model might end up training fine but it won't ignore the padding token in the labels which might impair learning? 

### Expected behavior

A bit more clear and concise context on what exact format preprocessing expects each model for each equivalent task.
And, why some models don't implement `ignore_index` while other do, since to my naive understanding if one uses `DataCollatorForSeq2Seq` then it will pad the labels automatically with `-100` if they are different seq len.

Am I missing something here?
